File: profil.py/felmeddelande.js

// Validera e-post vid inlogg
function validateLogin(email, password) {
    // Kolla om e-posten är i ogiltigt format
    if (!isEmailValid(email)) {
      return "Ogiltig e-postadress";
    }
  
    // Kolla om e-posten existerar i databasen/textfilen
    if (!isEmailExists(email)) {
      return "E-posten existerar inte i databasen";
    }
  
    // Kolla om lösenordet matchar angiven e-post
    if (!isPasswordCorrect(email, password)) {
      return "Fel lösenord för angiven e-postadress";
    }
  
    // Inloggning lyckades, ingen felmeddelande behövs
    return "";
  }
  
  // Funktion för att kolla om e-post är i godkänt format
  function isEmailValid(email) {
    // Implementera valideringslogik här (t.ex. med hjälp av reguljära uttryck)
    return true; // Returnera true om e-posten är i korrekt format, annars false
  }
  
  // Funktion för att kolla om e-posten existerar i databasen/textfilen
  function isEmailExists(email) {
    // Implementera söklogik här för att kolla om e-posten finns i databasen/textfilen
    return true; // Returnera true om e-posten existerar, annars false
  }
  
  // Funktion för att kolla om lösenordet tillhör angiven e-post
  function isPasswordCorrect(email, password) {
    // Implementera logik för att kolla om lösenordet matchar angiven e-post
    return true; // Returnera true om lösenordet är korrekt, annars false
  }
  
  // Använda funktionen vid inlogg
  let errorMessage = validateLogin("example@example.com", "password123");
  if (errorMessage) {
    // Visar felmeddelandet om det finns ett fel
    console.log(errorMessage);
  } else {
    // Ingen felmeddelande visas vid lyckad inloggning
    console.log("Inloggning lyckades");
  }
  
  
  

File: profil.py/felmeddelande.py
def check_login(email, password):
    # Kolla om e-posten är i giltigt format
    if not validate_email_format(email):
        return "Ogiltig e-postadress"

    # Kolla om e-posten existerar i databasen eller textfilen
    if not email_exists_in_database(email):
        return "E-postadressen existerar inte"

    # Kolla om lösenordet matchar e-postadressen
    if not verify_password(email, password):
        return "Fel lösenord"

    # Inloggningen lyckades
    return "Inloggning lyckades"


def validate_email_format(email):
    # Implementera validering av e-postadressens format här
    # Till exempel: använd en reguljär uttryckning för att kolla om det är ett giltigt e-postformat
    pass


def email_exists_in_database(email):
    # Implementera logik för att kolla om e-postadressen finns i databasen eller textfilen här
    pass


def verify_password(email, password):
    # Implementera logik för att verifiera lösenordet för den angivna e-postadressen här
    pass


# Användning av funktionen
email = input("Ange din e-postadress: ")
password = input("Ange ditt lösenord: ")

result = check_login(email, password)
print(result)

